panelapp/panelapp_utils.py

```diff
@@ -105,16 +109,22 @@ def _get_all_panels(panels_url, all_results):
         return _get_all_panels(next_page, all_results)
 
 
-def _get_all_genes_for_panel(panel_genes_url, all_results):
-    resp = requests.get(panel_genes_url)
+def _get_all_genes(genes_url: str, results_by_panel_id: dict):
+    resp = requests.get(genes_url, timeout=REQUEST_TIMEOUT_S)
     resp_json = resp.json()
-    all_results += resp_json.get('results', [])
+
+    for result in resp_json.get('results', []):
+        if result.get('panel'):
+            panel_id = result['panel']['id']
+            if panel_id not in results_by_panel_id:
```

Comment:
you do not have to change this, but we use `defaultdict` in a lot of places in the seqr codebase to avoid having to handle this edge case
```
results_by_panel_id = defaultdict(list)
...
results_by_panel_id[panel_id].append(result). // no need to check if key is present
```

Reply:
I updated the code to use `defaultdict`!